[Weave] New Weave projects page (#1913)
wandb/docs: weave/guides/platform/weave-projects.mdx

    - **[Scorers](/weave/guides/evaluation/scorers)**: Functions that evaluate model inputs and outputs to measure quality, safety, or custom metrics.
    - **[Ops](/weave/guides/tracking/ops)**: Automatically versioned and tracked functions decorated with `@weave.op()` that produce calls and maintain a record of function modifications.
    - **Other objects and metadata**: Additional Weave objects, custom types, and associated metadata stored within the project.

## Project paths

A Weave project is owned and managed by a [team](/platform/hosting/iam/org_team_struct#team) (also called an entity) in your W&B organization. Your team name and project name make up the path to your project in the W&B UI. You use also use this path in your code to associate your Weave resources with the project.

For example, if your team's name is `my-team` and your project's name is `my-project`, the path to your project is:

```
my-team/my-project
```

Use this path when you initialize Weave in your code to associate your app's data with the project:

```
weave.init('my-team/my-project')
```

> 

Reviewer: here's my suggestion: happy to chat this out if my reasoning isn't clear, but i have confusion calling that thing a 'sidebar' when it is not persistent).
I think we need a section to highlight the global navigation.  we'd discussed how that clickable logo isn't obvious. (it really wasn't to me)  I'd have put this section 'after' creating, but I see you reference the left sidebar there ... and I'm presuming that is what you mean ( and then I'd update the references of 'sidebar' in the other procedures to say **Global navigation menu**.)

## Find your projects
Click the **Weights & Biases logo** in the upper-left corner to open the **Global navigation menu**.
The menu lists your most recent projects and the teams you belong to, allowing you to quickly switch context.

Alternatively, open your **Profile**  from the menu under your name in the upper-right corner.  The **Projects** tab lists all of your projects.

4. After configuring the necessary fields, select **Create project**. The project is created and added to the top of the **Projects** list in the left sidebar. 

When you open the new project, you see quickstarts for both W&B Models and W&B Weave. The sidenav contains tools for each product. After you associate Weave data with your project through your code, the sidenav shows only Weave tools and the Weave dashboard becomes available.
</Tab>

<Tab title="Using code">

You can create new Weave projects directly from your code. To do this, add a new project path as an argument to `weave.init()`:

<CodeGroup>
```python
import weave

weave.init('my-team/my-llm-app')
```

```typescript
import * as weave from 'weave';

await weave.init('my-team/my-llm-app');
```
</CodeGroup>

When you run the code, Weave creates the new project in your W&B account and uploads any applicable data from the run to the project. Weave also configures the project as a Weave project (not a W&B Models project), and the UI sidenav shows only Weave tools.

</Tab>

</Tabs>

## Viewing projects in the UI

To access a project in the Weave UI:

1. Navigate to [wandb.ai](https://wandb.ai).

> 2. Select your project from the **Projects** sidebar. This opens the project's overview page, or **Traces** page if you have already logged traces to the project.

Reviewer: for example this isn't clear to me 'Select your project from the **Projects** sidebar.'
The me from a month ago couldn't have followed this :).    I think you are talking about (what i'm calling) the global navigation menu, and once that's introduced, hopefully that is clear.    that, or clarify this some other way

---
title: "Work with Weave Projects"
description: "Use Weave projects to organize related assets like traces, prompts, evaluations, models, and dashboards."
---

> Weave projects are organizational containers that group related traces, evaluations, models, datasets, and other Weave assets together. Projects help you organize your work, collaborate with team members, and track the evolution of your LLM applications over time.

Reviewer 2: ```suggestion
Weave projects keep related assets organized, including traces, evaluations, models, datasets, and other Weave assets. Projects keep your team organized, streamline collaboration and sharing, and help you track the evolution of your LLM applications over time.
```
WDYT? I keep trying to make this more active. Totally optional.

Projects store the following information and resources:

> - **[Traces](/weave/guides/tracking/tracing)**: Individual LLM calls, function executions, and application flows that capture inputs, outputs, latency, and token usage for debugging and analysis.

Reviewer 2: I find this list a bit overwhelming on the page. I wonder if it could be presented in a different format than nested lists.

- **Assets**: The Assets tab in the Weave UI stores and organizes the following resources in your project:
    - **[Models](/weave/guides/core-types/models)**: Versioned model definitions that capture prompts, parameters, and configuration, automatically tracking changes for reproducibility and comparison.
    - **[Datasets](/weave/guides/core-types/datasets)**: Collections of test examples stored as Weave objects that can be downloaded, browsed, and used for running evaluations.
    - **[Prompts](/weave/guides/core-types/prompts)**: Versioned system prompts that you can test and reuse in your model configurations.
    - **[Scorers](/weave/guides/evaluation/scorers)**: Functions that evaluate model inputs and outputs to measure quality, safety, or custom metrics.
    - **[Ops](/weave/guides/tracking/ops)**: Automatically versioned and tracked functions decorated with `@weave.op()` that produce calls and maintain a record of function modifications.
    - **Other objects and metadata**: Additional Weave objects, custom types, and associated metadata stored within the project.

## Project paths

> A Weave project is owned and managed by a [team](/platform/hosting/iam/org_team_struct#team) (also called an entity) in your W&B organization. Your team name and project name make up the path to your project in the W&B UI. You use also use this path in your code to associate your Weave resources with the project.

Reviewer 2: I haven't landed on the exact way to do it but I think this could be more active too. The team does the owning, and the team name determines the other details, right?

For example, if your team's name is `my-team` and your project's name is `my-project`, the path to your project is:

```
my-team/my-project
```

Use this path when you initialize Weave in your code to associate your app's data with the project:

```
weave.init('my-team/my-project')
```

## Create and find projects

You can create a Weave project using the W&B UI or by setting a new project path in your code.

<Tabs>

<Tab title="Using the UI">
To create a new project using the W&B UI:

1. Navigate to [wandb.ai](https://wandb.ai).
2. Select **New project**. This opens the **Create a new project** modal dialog.
3. In the modal, configure the following fields:
    * **Team**: The existing team to associate the project with.
    * **Name**: A name for your project.
    * **Project visibility**: The scope of who can view and edit the project.
    * **Description**: (Optional) A short description of the project to help you track and organize its purpose.
4. After configuring the necessary fields, select **Create project**. The project creates the project.

> ### Find your project

Reviewer 2: This might be better as "Find a project"

To locate your newly created project, select the **Weights & Biases logo** in the upper-left corner to open the **Global navigation** menu. The menu lists your most recent projects and the teams you belong to, allowing you to quickly switch context. Select your project to open the project's workspace.

Alternatively, open your **Profile** from the menu under your name in the upper-right corner. The **Projects** tab lists all of your projects.

When you open the new project's workspace, the UI offers quickstarts for both W&B Models and W&B Weave. The project's toolbar also contains tools for each product. After you associate Weave data with your project through your code, the sidenav shows only Weave tools and the Weave dashboard becomes available.

</Tab>

> <Tab title="Using code">

Reviewer 2: I don't love the gerunds in tab titles. Have we used this pattern elsewhere?

> You can create new Weave projects directly from your code. To do this, add a new project path as an argument to `weave.init()`:

Reviewer 2: ```suggestion
To create a Weave project directly from your code, provide a new project path as an argument to `weave.init()`:
```

<CodeGroup>
```python
import weave

weave.init('my-team/my-llm-app')
```

```typescript
import * as weave from 'weave';

await weave.init('my-team/my-llm-app');
```
</CodeGroup>

When you run the code, Weave creates the new project in your W&B account and uploads any applicable data from the run to the project. Weave also configures the project as a Weave project (not a W&B Models project), and the UI sidenav shows only Weave tools.

</Tab>

</Tabs>

> ## Viewing projects in the UI

Reviewer 2: ```suggestion
## View projects in the UI
```

---
title: "Weave projects overview"

> description: "Weave projects are collections of traces, evaluations, models, and other Weave assets"

Reviewer 2: How about something like "Use Weave projects to organize related assets like traces, prompts, evaluations, models, and dashboards"? FYI the description also shows up at the top of the page content, so you may not need to duplicate it all in the first paragraph, depending on what you think about the preview.

- **Assets**: The Assets tab in the Weave UI stores and organizes the following resources in your project:
    - **[Models](/weave/guides/core-types/models)**: Versioned model definitions that capture prompts, parameters, and configuration, automatically tracking changes for reproducibility and comparison.
    - **[Datasets](/weave/guides/core-types/datasets)**: Collections of test examples stored as Weave objects that can be downloaded, browsed, and used for running evaluations.
    - **[Prompts](/weave/guides/core-types/prompts)**: Versioned system prompts that you can test and reuse in your model configurations.
    - **[Scorers](/weave/guides/evaluation/scorers)**: Functions that evaluate model inputs and outputs to measure quality, safety, or custom metrics.
    - **[Ops](/weave/guides/tracking/ops)**: Automatically versioned and tracked functions decorated with `@weave.op()` that produce calls and maintain a record of function modifications.
    - **Other objects and metadata**: Additional Weave objects, custom types, and associated metadata stored within the project.

> ## Creating and using projects

Reviewer 2: Aha, so there are tasks here after all, so maybe "Projects" is the right title for the page.

You could call this section "Work with projects", then you could have Create, View, Manage, Delete. Or you could have a section Projects with "Overview" and "Work with projects" or something like that. I'd prefer the non-gerund forms.

> Weave projects exist within [teams](/platform/hosting/iam/org_team_struct#team) (also called entities) in your W&B organization. The full project path follows this format:

Reviewer 2: ```suggestion
A Weave project is owned and managed by a [team](/platform/hosting/iam/org_team_struct#team) (also called an entity) in your W&B organization. The full project path follows this format:
```
I wonder if it would be better to use inline code for the formatting. This isn't something that your reader would copy and paste.

```
team-name/project-name
```

For example, a team called `my-team` with a project called `llm-chatbot` has the project path `my-team/llm-chatbot`.

> To associate traces, evaluations, or any other resources with a project, you must pass the project's path as a string when you initialize Weave in your code, like this:

Reviewer 2: ```suggestion
To associate traces, evaluations, and other resources with a project, pass the project's path as a string when you initialize Weave in your code:
```

<Tabs>
    <Tab title="Python">
    ```python
    import weave

    weave.init('my-team/my-llm-app')
    ```
    </Tab>
    <Tab title="TypeScript">
    ```typescript
    import * as weave from 'weave';

    await weave.init('my-team/my-llm-app');
    ```
    </Tab>
</Tabs>

If you don't specify a team name, Weave uses your [default entity](/platform/app/settings-page/user-settings#default-team) (your personal account or default team).

## Viewing projects in the UI

> Access your projects in the Weave UI:

Reviewer 2: ```suggestion
Access a project in the Weave UI:
```

> 1. Navigate to [wandb.ai](https://wandb.ai)

Reviewer 2: ```suggestion
1. Navigate to [wandb.ai](https://wandb.ai).
```
FWIW I tend to copy what Noah does: https://docs.wandb.ai/models/registry/create_registry

> 2. Select your project from the **Projects** sidebar. This opens the project's overview page.

Reviewer 2: You could send them straight here in step 1.

Reviewer: I like step 1, imo.   But for step 2:
2. From the sidebar menu, click **Project**. This opens the project’s overview page.

I 'think' this 'might' be what we are standardizing towards ;) based on the slack conversation.  in other places now I'd say 'weave sidebar'...but in this case it doesn't matter since project appears in both.

i also think maybe we should add maybe a Note somewhere like:
The W&B sidebar menu changes depending on the activity within your project.  Your sidebar displays different options if you are doing Models work or Weave work in your project.  A fresh project displays both, but might later change to display one or the other depending on your project’s usage.

3. From the overview page, you can: 
    * Review your projects's statistics, such as the total number of traces logged to the project and how much storage the project is using.
    * Navigate to traces, evaluations, and other resources using the sidebar.
    * Navigate to recently logged traces, evaluations, models, and datasets.

<Frame>

> ![Weave project](imgs/weave-project.png)

Reviewer 2: ```suggestion
![Weave project](/weave/guides/platform/imgs/weave-project.png)
```
